WordPress/src/main/java/org/wordpress/android/ui/accounts/signup/SettingsUsernameChangerFragment.kt

```diff
@@ -0,0 +1,193 @@
+                isEnabled = false
+            }
+
+            // Toggles the enabled property of the button based on the correctness of the username being typed.
+            usernameControl.addTextChangedListener(object : TextWatcher {
+                override fun afterTextChanged(editable: Editable?) {
+                    editable?.let {
+                        changeUsernameButton.isEnabled = (it.toString() == username)
+                    }
+                }
+
+                override fun beforeTextChanged(s: CharSequence?, start: Int, count: Int, after: Int) {
+                }
+
+                override fun onTextChanged(s: CharSequence?, start: Int, before: Int, count: Int) {
+                }
+            })
+            usernameControl.requestFocus()
+        }
+    }
+
+    private fun saveUsername(username: String) {
+        showProgress()
+        val payload = PushUsernamePayload(
+                username, KEEP_OLD_SITE_AND_ADDRESS
+        )
+        mDispatcher.dispatch(AccountActionBuilder.newPushUsernameAction(payload))
+    }
+
+    @Subscribe(threadMode = ThreadMode.MAIN)
+    fun onUsernameChanged(event: OnUsernameChanged) {
+        if (event.isError) {
+            AnalyticsTracker.track(ACCOUNT_SETTINGS_CHANGE_USERNAME_FAILED)
+            AppLog.e(
+                    T.API, "SettingsUsernameChangerFragment.onUsernameChanged: " +
+                    event.error.type + " - " + event.error.message
+            )
+            endProgress()
+            showErrorDialog(SpannableStringBuilder(getString(R.string.signup_epilogue_error_generic)))
+        } else if (event.username != null) {
+            AnalyticsTracker.track(ACCOUNT_SETTINGS_CHANGE_USERNAME_SUCCEEDED)
+            endProgress()
+            val result = Bundle().apply { putString(RESULT_USERNAME, event.username) }
+            dialogController.confirm(result)
+        }
+    }
+
+    fun showProgress() {
+        if (progressDialog == null || progressDialog?.window == null || progressDialog?.isShowing == false) {
+            progressDialog = ProgressDialog(context).apply {
+                isIndeterminate = true
+                setCancelable(true)
+                setMessage(getString(R.string.settings_username_changer_progress_dialog))
+                setOnCancelListener { showActionCancelMessage() }
+            }
+        }
+        try {
+            if (progressDialog?.isShowing == false) {
+                progressDialog?.show()
+            }
+        } catch (e: WindowManager.BadTokenException) {
+            AppLog.e(T.SETTINGS, e)
+        }
+    }
+
+    private fun endProgress() {
+        if (progressDialog != null && progressDialog?.isShowing == true && progressDialog?.window != null) {
+            progressDialog?.dismiss()
+        }
+        progressDialog = null
+    }
+
+    private fun showActionCancelMessage() {
+        val duration = AccessibilityUtils.getSnackbarDuration(
```

Comment:
It seems you used a resource from Site Creation `site_creation_snackbar_duration ` was that intentional?  I think we might want to consider using `Snackbar.LENGTH_LONG` and perhaps even using `WPSnackBar` instead of `WPDialogSnackbar`. 

This is from a post from Tyler who created `WPSnackBar` and  `WPDialogSnackbar` -> `If you need a title, long message, or multiple action buttons, then use WPDialogSnackbar.  If you need a message with one or two lines of text and zero or one action button, then use WPSnackbar.`

Perhaps, the message is long and using the `WPDialogSnackbar` make sense. I'll leave the final decision to you.

Wdyt?

Reply:
Fixed in https://github.com/wordpress-mobile/WordPress-Android/pull/10436/commits/01c8d819703b7a848b1c702ad46b98e59d03ce9e